api/src/indicators.py

```diff
@@ -486,10 +478,21 @@ def patch_annotation(payload: MetadataSchema.MetaModel, indicator_id: str):
 
         body = json.loads(payload.json(exclude_unset=True))
 
+        # Handles datasets being regsitered with no date.
+        if not body.get("annotations").get("date", []):
```

Comment:
yea I'll split this out to check if "annotations" exists. The second get will fail here since the first will default to None and calling .get() on a None type won't work.